src/Access/AccessControlManager.cpp

```diff
@@ -295,6 +308,12 @@ void AccessControlManager::addStoragesFromUserDirectoriesConfig(
             bool readonly = config.getBool(prefix + ".readonly", false);
             addDiskStorage(name, path, readonly);
         }
+        else if (type == LDAPAccessStorage::STORAGE_TYPE)
+        {
+            if (bracket_pos != String::npos)
+                throw Exception("Duplicate storage type '" + type + "' at " + prefix + " in config", ErrorCodes::EXCESSIVE_ELEMENT_IN_CONFIG);
```

Comment:
With the new `IAccessStorage::login()` function I think this deduplication can be made weaker because there is no reason to prohibit having two `LDAPAccessStorage`s at a time.

Reply:
Removed that restriction (and the code that enforces).